src/task/collection.coffee

```diff
@@ -7,6 +7,20 @@ exercises = require '../exercise/collection'
 tasks = {}
 channel = new EventEmitter2 wildcard: true
 
+ERRORS_TO_SILENCE = ['page_has_no_exercises']
+
+getUnhandledErrors = (errors) ->
+  otherErrors = _.reject errors, (error) ->
+    _.indexOf(ERRORS_TO_SILENCE, error.code) > -1
+
+handledAllErrors = (otherErrors) ->
+  _.isEmpty otherErrors
+
+checkFailure = (response) ->
+  if response.data.errors
+    response.data.errors = getUnhandledErrors(response.data.errors)
+    response.stopErrorDisplay = handledAllErrors(response.data.errors)
+
 load = (taskId, data) ->
```

Comment:
I think we should move this to a helper, and set the errors to silence inside the various modelly-collectionny things, after this is merged